nbgrader2canvas.py

```diff
@@ -36,6 +70,17 @@
     else:
         source = source[0]
     score = float(items[7])
+
+    try:
+        previous_upload_hash = upload_history.loc[(canvas_id, assignment_name), UPLOAD_LOG_COLUMNS[2]]
+    except KeyError:
+        previous_upload_hash = 0
+    source_hash = sha256(str(parse_notebook_as_json(source)))
+    if (source_hash == previous_upload_hash):
+        print("{}'s autograded notebook is (functionally) identical to the previously uploaded notebook, skipping.".format(canvas_id))
```

Comment:
The wording is a little long, but specifying `functionally` could help with debugging if any unforeseen issues pop up.